### ISSUE TITLE
Windows os.path.isabs UNC path bug

### ISSUE THREAD

**Issue by 3912e9d3-82a6-4b5f-91d7-76e05b474b45** [python/cpython#66498]
BPO | [22302](https://bugs.python.org/issue22302)
--- | :---
Nosy | @pitrou, @serhiy-storchaka, @eryksun, @zooba

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2014-08-29.21:58:22.504>
labels = ['3.10', 'type-bug', 'library', '3.9', 'OS-windows']
title = 'Windows os.path.isabs UNC path bug'
updated_at = <Date 2021-02-25.17:05:14.388>
user = 'https://bugs.python.org/akima'
```

bugs.python.org fields:
```python
activity = <Date 2021-02-25.17:05:14.388>
actor = 'eryksun'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)', 'Windows']
creation = <Date 2014-08-29.21:58:22.504>
creator = 'akima'
dependencies = []
files = []
hgrepos = []
issue_num = 22302
keywords = []
message_count = 18.0
messages = ['226091', '226108', '226122', '226243', '226427', '226429', '226430', '226432', '226434', '226436', '226440', '226444', '226455', '226473', '226475', '226477', '285510', '387681']
nosy_count = 5.0
nosy_names = ['pitrou', 'serhiy.storchaka', 'eryksun', 'steve.dower', 'akima']
pr_nums = []
priority = 'normal'
resolution = None
stage = 'needs patch'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue22302'
versions = ['Python 3.9', 'Python 3.10']
```

</p></details>

**Comment by zooba** [comment]
> Would a new _strict_ argument for `isabs()` help?

No, we should just strive for the most consistency. Testing for starting with two sep/altseps or `{A-Z}:{sep}` should be enough. And it should be consistent with how `join()` handles each segment.

If users really want to know whether a path is relative but has a drive segment, they can `splitdrive`.

**Comment by barneygale** [comment]
I think I misunderstood this bug, and thought it was a dupe of #44626.

This bug appears to be fixed in 3.12 by #100351:

```python
>>> import ntpath
>>> ntpath.isabs(r"\\server")
True
>>> ntpath.isabs(r"\\server\share")
True
>>> ntpath.isabs(r"\\server\share\folder")
True
>>> ntpath.isabs(r"\\server\share\folder\folder")
True
```

**Comment by zooba** [comment]
Yeah, it was fixed by that, but the fix then caused #112563 (which is already being looked into). So this one can be closed.
